### PR TITLE
feature: add extra containers sections

### PR DESCRIPTION
**Pull request by Moon1706**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?

This merge request adds extra init and sidecar containers in bank-vault pod. It may be useful when you sidecar containers e.g. consul-agent. 

### Why?

That helps DevOps engineers customize bank-vault. These sections are used in the official Vault chart https://github.com/hashicorp/vault-helm.

### Additional context

This MR tested with `helm template .`

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: charts/vault/README.md [hunk 1]
```diff
@@ -140,6 +140,9 @@ The following tables lists the configurable parameters of the vault chart and th
 | `tolerations`                        | List of node tolerations for the pods. https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/  | `[]`                                                                                            |
 | `affinity`                           | Node affinity settings for the pods. https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/   | If not set it defaults to`podAntiAffinity` to `preferredDuringSchedulingIgnoredDuringExecution` |
 | `labels`                             | Additonal labels to be applied to the Vault StatefulSet and Pods                                                | `{}`                                                                                            |
+| `extraInitContainers`                | Init containers to add to the bank-vault pod              | `[]`                                                                                            |
```
Comment (pbalogh-sa) [inline charts/vault/README.md:143]: pleas fix bank-vault to bank-vaults

File: charts/vault/README.md [hunk 2]
```diff
@@ -140,6 +140,9 @@ The following tables lists the configurable parameters of the vault chart and th
...
+| `extraContainers`                    | Sidecar containers to add to the bank-vaul pod            | `[]`                                                                                            |
```
Comment (pbalogh-sa) [inline charts/vault/README.md:144]: pleas fix bank-vaul to bank-vaults

**Review by pbalogh-sa — CHANGES_REQUESTED** [review]

File: charts/vault/README.md (hunk 1 above)
Comment (Moon1706) [reply charts/vault/README.md]: Done

File: charts/vault/README.md (hunk 2 above)
Comment (Moon1706) [reply charts/vault/README.md]: Done

**Review by pbalogh-sa — APPROVED** [review]

**Pull request merged**